modules/seqexec/engine/src/main/scala/seqexec/engine/Engine.scala

```diff
@@ -49,6 +49,22 @@ class Engine[D, U](stateL: Engine.State[D]) {
       case None      => unit
     }
 
+  /*
+   * startFrom starts a sequence from an arbitrary step. It does it by marking all previous steps to be skipped and then
+   * modifying the state sequence as if it was run.
+   * If the requested step is already run or marked to be skipped, the sequence will start from the next runnable step
+   */
+  def startFrom(id: Observation.Id, step: StepId): HandleType[Unit] =
```

Comment:
Another option would have been to implement it as a sequence of events, like MarkSkip(sid, 1), MarkSkip(sid, 2),..,Start(sid), but there is the risk that another event is processed in between. As it is, everything is resolved in one iteration of the engine.

Reply:
Good call